lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -291,17 +296,13 @@ def initconfig_hardware_entries(self):
                 if os.path.basename(os.path.normpath(llvm_prefix)) != "llvm":
                     llvm_bin = os.path.join(llvm_prefix, "llvm/bin/")
                 entries.append(
-                    cmake_cache_filepath(
-                        "CMAKE_HIP_COMPILER", os.path.join(llvm_bin, "amdclang++")
-                    )
+                    cmake_cache_filepath("CMAKE_HIP_COMPILER", os.path.join(llvm_bin, "clang++"))
                 )
 
-            archs = self.spec.variants["amdgpu_target"].value
-            if archs[0] != "none":
-                arch_str = ";".join(archs)
-                entries.append(cmake_cache_string("CMAKE_HIP_ARCHITECTURES", arch_str))
-                entries.append(cmake_cache_string("AMDGPU_TARGETS", arch_str))
-                entries.append(cmake_cache_string("GPU_TARGETS", arch_str))
+                if archs[0] != "none":
+                    arch_str = ";".join(archs)
+                    entries.append(cmake_cache_string("AMDGPU_TARGETS", arch_str))
```

Comment:
So this will need to be set for non BLT < 0.6 targets, right?  Right now for other codes that don't use BLT, it looks like you have cut these out of the build.

Reply:
I don't think `AMDGPU_TARGETS` was ever a real CMake variable.

Reply:
nm it was and is now deprecated:

https://github.com/ROCm/rocBLAS/blob/6dc0ae484c64529a6dc3de7d5da4ebeeb34eaf13/CMakeLists.txt#L143-L144

Reply:
Thank you for the pointer @white238 !

It makes me think that @bvanessen is making a point regarding `GPU_TARGET` which I also removed from the general case. It looks like I should restore it (sounds like I need a rocblas scenario to catch this).

Regarding `AMDGPU_TARGET`, it is still defined in the rocBLAS case to ensure backward compatibility.

Takeaways:
- I am going to keep ensuring backward compatibility in the general case by defining all three
- I am not going to define `AMDGPU_TARGET` in blt@0.7: case.
- @white238 don't you think, per the rocBLAS example, that BLT HIP setup should be setting `GPU_TARGET` based on the value of `CMAKE_HIP_ARCHITECTURE` ?

Reply:
`GPU_TARGETS` and `AMDGPU_TARGETS` seem to be project specific CMake variables in several rocM projects:

https://sourcegraph.com/search?q=repo:github.com/rocm/*+GPU_TARGETS&patternType=literal&sm=0

But unless you are including those CMake projects in your CMake projects, we do not need to set them. I searched all of Kitware's git repos and neither are referenced.

Reply:
@bvanessen with the current state, this PR is not affecting hip handling for lbann anymore.